### DISCUSSION TITLE
Apache Age support, A Graph extension for postgresql.

### DISCUSSION THREAD

**Discussion by sumitsharansatsangi** [piccolo-orm/piccolo#577] [Ideas]
Apache Age is a graph extension for postgresql.
For more info, visit [https://age.apache.org/docs/](https://age.apache.org/docs/)

Using psycopg2, there is an driver for python .
For more info visit [https://pypi.org/project/apache-age-python/](https://age.apache.org/docs/)

Future is with the Graph and async. 
So, it will be great if piccolo can support this .

This feature can make piccolo of the next level.

**Comment by sinisaos** [comment]
@sumitsharansatsangi I don't know anything about Apache AGE but Piccolo uses [asyncpg](https://piccolo-orm.readthedocs.io/en/latest/piccolo/deployment/index.html?highlight=asyncpg#base-image) database driver and Apache AGE requirements are [Psycopg2 ](https://github.com/apache/age/tree/master/drivers/python#prerequisites) driver so I think for now it is not possible.

**Reply by sumitsharansatsangi** [reply]
Sorry, Not known that it supports only Psycopg2. 

Thanks for quick response and contributing to the nice project.

**Reply by sinisaos** [reply]
No worries and thanks for the kind words. [There](https://github.com/apache/age/issues/43) is issue about asyncpg but as I understand there is limited support. I have no experience with GraphQL and maybe @dantownsend  would know more about it, but if you need GraphQL, maybe you can try [this](https://testdriven.io/blog/fastapi-graphql/). It uses Orator ORM, but you can modify it to use Piccolo ORM and it works. Piccolo ORM has a really neat feature [create_pydantic_model](https://piccolo-orm.readthedocs.io/en/latest/piccolo/serialization/index.html#create-pydantic-model) that can help you easily create Pydantic models from Piccolo Table and then use it in [previous example](https://testdriven.io/blog/fastapi-graphql/).
